pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -632,3 +640,23 @@ func (r *Reconciler) matchControllerRequirement(app *v1beta1.Application) bool {
 	}
 	return true
 }
+
+const (
+	// ComponentNamespaceContextKey is the key in context that defines the override namespace of component
+	ComponentNamespaceContextKey contextKey = iota
```

Comment:
will it be overriden by `context[0] = "other-thing"`

Reply:
No. There is no int key 0. It will be pkg/controller/core.oam.dev/v1alpha2/application:contextKey(0).